### DISCUSSION TITLE
How do I send get params when invoking a function that uses the GET method?

### DISCUSSION THREAD

**Discussion by alita-moore** [supabase/supabase#19816] [General]
I see in the docs there's an example of delete sending details in the body. 
```ts
const { data, error } = await supabase.functions.invoke('hello', {
  headers: {
    "my-custom-header": 'my-custom-header-value'
  },
  body: { foo: 'bar' },
  method: 'DELETE'
})
```
but this seems weird to me because DELETE doesn't have a request body, no? So does that mean that the "body" property is also used as the params object in the case of a GET / DELETE request?

**Comment by GaryAustin1** [comment]
Both DELETE and GET allow a body BUT it is not defined to require it.  So it is up to the destination if it will handle it or not.
For instance Supabase storage delete handles a body for bulk delete.
![image](https://github.com/supabase/supabase/assets/54564956/5a1fde79-ef3a-4055-8c43-d6cd46f60e4b)

**Reply by alita-moore** [reply]
so then there's no param option?

**Reply by GaryAustin1** [reply]
It looks like from invoke you pass all parameters in the body for all methods.
You can see the source here:
https://github.com/supabase/functions-js/blob/main/src/FunctionsClient.ts
And it just changes the verb of the http call with the rest the same for all.

**Reply by GaryAustin1** [reply]
If your function is expecting the parameters in the URL then you would likely have to call the function with javascript fetch and provide the URL with parameters and the Authorization header if needed.

**Reply by alita-moore** [reply]
what's the benefit of that? why not just include the option to include parameters? including it in the body goes against the specs, I thought.

**Reply by GaryAustin1** [reply]
I don’t know why they did it that way.  It is not against the specs as it is allowed, you just can’t count on it being handled, so can only do that if the endpoint handles it.  

You can create an issue in function-js asking for an enhancement.

**Comment by justadev-afk** [comment]
> I see in the docs there's an example of delete sending details in the body.
> 
> ```ts
> const { data, error } = await supabase.functions.invoke('hello', {
>   headers: {
>     "my-custom-header": 'my-custom-header-value'
>   },
>   body: { foo: 'bar' },
>   method: 'DELETE'
> })
> ```
> 
> but this seems weird to me because DELETE doesn't have a request body, no? So does that mean that the "body" property is also used as the params object in the case of a GET / DELETE request?

i have found a solution for this, just pass the query params on the function name, like this
```typescript
await client.functions.invoke(`${FUNCTION_NAME}?id=${id}`, {
  method: "DELETE"
});
```

**Reply by joshdance** [reply]
This was helpful. 

`invoke(`${FUNCTION_NAME}?id=${id}``

I tried all the ways above and this is the only one that worked to actually send in URL params for me.
